tiledb/api/src/array/attribute/pod.rs

```diff
@@ -0,0 +1,105 @@
+use tiledb_common::array::CellValNum;
+use tiledb_common::datatype::Datatype;
+use tiledb_common::filter::FilterData;
+use tiledb_common::{metadata_value_go, physical_type_go};
+use tiledb_pod::array::attribute::{AttributeData, FillData};
+
+use super::{Attribute, Builder};
+use crate::error::Error as TileDBError;
+use crate::{Context, Factory, Result as TileDBResult};
+
+impl TryFrom<&Attribute> for AttributeData {
+    type Error = TileDBError;
+
+    fn try_from(attr: &Attribute) -> Result<Self, Self::Error> {
+        let datatype = attr.datatype()?;
+        let fill = physical_type_go!(datatype, DT, {
+            let (fill_value, fill_value_nullability) =
+                attr.fill_value_nullable::<&[DT]>()?;
+            FillData {
+                data: fill_value.to_vec().into(),
+                nullability: Some(fill_value_nullability),
+            }
+        });
+
+        Ok(AttributeData {
+            name: attr.name()?,
+            datatype,
+            nullability: Some(attr.is_nullable()?),
+            cell_val_num: Some(attr.cell_val_num()?),
+            fill: Some(fill),
+            filters: Vec::<FilterData>::try_from(&attr.filter_list()?)?,
+        })
+    }
+}
+
+impl TryFrom<Attribute> for AttributeData {
```

Comment:
I assume this was just moved from wherever it was before, but seeing a try_from for both a ref and consuming seems odd. I can't quite convince myself this version would never be needed, but it sure seems unlikely.

Not asking for a change here. I'm just noting something I saw for future me writing trait implementations.

Reply:
I don't really remember why I added this the first time, yeah.  Probably trying to do something mildly clever with iterators.